staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -663,6 +674,19 @@ func forceUmount(path string) error {
 	out, cmderr := cmd.CombinedOutput()
 
 	if cmderr != nil {
+		return fmt.Errorf("unmount failed: %v\nUnmounting arguments: -f %s\nOutput: %s", cmderr, path, string(out))
+	}
+	return nil
+}
+
+func quietUmount(path string) error {
```

Comment:
Won't this break distroless environments?
Is it possible to use a syscall.Unmount with MNT_DETATCH with lazy unmounting.

Ref: https://man7.org/linux/man-pages/man2/umount.2.html
```
       MNT_DETACH (since Linux 2.4.11)
              Perform a lazy unmount: make the mount unavailable for new
              accesses, immediately disconnect the filesystem and all
              filesystems mounted below it from each other and from the
              mount table, and actually perform the unmount when the
              mount ceases to be busy.
```

As long as kubelet can handle file descriptors properly, I believe there should be no issues then and will give a fast path.

Reply:
Sorry! The code already relies on util-linux; still will be nice to get rid of it.

Reply:
ya I am assuming that umount == util-linux umount which must be printing 'not mounted' error. https://github.com/util-linux/util-linux/blob/v2.2/mount/umount.c#L179 and at least on my vanilla amazon linux 2 Node,  which of course represents just 1 sample , it does

Reply:
I am afraid that lazy unmounting might cause os.Remove call afterward to fail if the unmounting didn't complete and dir is not empty

Reply:
we should call /bin/unmount to make sure that any umount helpers are called too. For in-tree volumes it's umount.nfs, CSI drivers may have their own.

Calling kernel directly could be OK only when the caller knows it's a bind-mount or knows that it does not need any helper..